pkg/infra/spot.go

```diff
@@ -65,14 +60,27 @@ var _ = Describe("[Feature:Machines] Running on Spot", func() {
 		}
 
 		By("Creating a Spot backed MachineSet", func() {
-			machineSetParams = framework.BuildMachineSetParams(client, machinesCount)
-			Expect(setSpotOnProviderSpec(platform, machineSetParams, "")).To(Succeed())
+			machineSetReady := false
+			machineSetParams := framework.BuildMachineSetParams(client, machinesCount)
+			machineSetParamsList := framework.BuildAlternativeMachineSetParams(machineSetParams, platform)
+			for _, machineSetParams := range machineSetParamsList {
+				Expect(setSpotOnProviderSpec(platform, machineSetParams, "")).To(Succeed())
 
-			machineSet, err = framework.CreateMachineSet(client, machineSetParams)
-			Expect(err).ToNot(HaveOccurred())
-			delObjects[machineSet.Name] = machineSet
-
-			framework.WaitForMachineSet(client, machineSet.GetName())
+				machineSet, err = framework.CreateMachineSet(client, machineSetParams)
+				Expect(err).ToNot(HaveOccurred())
+				delObjects[machineSet.Name] = machineSet
+
+				err = framework.WaitForSpotMachineSet(client, machineSet.GetName())
+				if err == framework.ErrMachineCreateFailedInsufficientCapacity {
+					// If machineSet cannot scale up due to insufficient capacity, try again with different machineSetParams
+					framework.WaitForMachineSetDelete(client, machineSet)
+					continue
+				}
```

Comment:
If we add a new `By()` in here we can add something to the test output when we have to recreate a new MachineSet. Will be helpful to identify when we have to iterate